src/transaction/hooks/useSendWalletTransactions.tsx

```diff
@@ -0,0 +1,72 @@
+import { useCallback } from 'react';
+import type { ContractFunctionParameters } from 'viem';
+import type { Config } from 'wagmi';
+import type {
+  SendTransactionMutateAsync,
+  WriteContractMutateAsync,
+} from 'wagmi/query';
+import type { WalletCapabilities as OCKWalletCapabilities } from '../../types';
+import {
+  TRANSACTION_TYPE_CALLS,
+  TRANSACTION_TYPE_CONTRACTS,
+} from '../constants';
+import type { Call, WalletCapabilities } from '../types';
+
+export const useSendWalletTransactions = ({
+  transactions,
+  transactionType,
+  capabilities,
+  writeContractsAsync,
+  writeContractAsync,
+  sendCallsAsync,
+  sendCallAsync,
+  walletCapabilities,
+}: {
+  transactions: Call[] | ContractFunctionParameters[];
+  transactionType: string;
+  capabilities?: WalletCapabilities;
+  // biome-ignore lint: cannot find module 'wagmi/experimental/query'
+  writeContractsAsync: any;
+  writeContractAsync: WriteContractMutateAsync<Config, unknown> | (() => void);
+  // biome-ignore lint: cannot find module 'wagmi/experimental/query'
+  sendCallsAsync: any;
+  sendCallAsync: SendTransactionMutateAsync<Config, unknown> | (() => void);
+  walletCapabilities: OCKWalletCapabilities;
```

Comment:
Let's create a `useSendWalletTransactionsReact` or `useSendWalletTransactionsInput` type to clean this up